tests/functional/u/unreachable.py

```diff
@@ -32,3 +36,46 @@ def func6():
     return
     yield
     print("unreachable")  # [unreachable]
+
+def func7():
+    sys.exit(1)
+    var = 2 + 2  # [unreachable]
+    print(var)
+
+def func8():
+    signal.signal(signal.SIGTERM, lambda *args: sys.exit(0))
+    try:
+        print(1)
+    except KeyboardInterrupt:
+        pass
+
+class FalseExit:
+    def exit(self, number):
+        print(f"False positive this is not sys.exit({number})")
+
+def func_false_exit():
+    sys  = FalseExit()
+    sys.exit(1)
+    var = 2 + 2
+    print(var)
+
+def func9():
+    os._exit()
+    var = 2 + 2  # [unreachable]
+    print(var)
+
+def func10():
+    exit()
+    var = 2 + 2  # [unreachable]
+    print(var)
+
+def func11():
+    quit()
+    var = 2 + 2  # [unreachable]
+    print(var)
+
+incognito_function = sys.exit
```

Comment:
I can't find the previous discussion, but to reply to one of your earlier questions: `another_exit` worked because it ends with `exit` 😄 Since this ends with `function` it shows we really do need that inference here.